src/mesh/RTree.cpp

```diff
@@ -24,10 +26,25 @@ rtree::PtrRTree rtree::getVertexRTree(PtrMesh mesh)
   return tree;
 }
 
+PtrPrimitiveRTree rtree::getPrimitiveRTree(PtrMesh mesh)
```

Comment:
Looks like you're doing essentially the same as in `getVertexRTree`. Why not use the same code? e.g. `emplace_back` ...

Reply:
You are right, will change the implementation to using `map emplace`.